editoast/src/views/layers/mvt_utils.rs

```diff
@@ -0,0 +1,256 @@
+            // Converts array as string
+            feature.add_tag_string(&tag_name, &serde_json::to_string(&json_values).unwrap())
+        }
+        JsonValue::Object(map_values) => {
+            for (key, value) in map_values.into_iter() {
+                add_tags_to_feature(
+                    feature,
+                    value,
+                    if tag_name.is_empty() {
+                        key
+                    } else {
+                        format!("{tag_name}_{key}")
+                    },
+                );
+            }
+        }
+        JsonValue::Null => (),
+    }
+}
+
+/// Creates a MVT tile and fills it with records
+///
+/// # Arguments
+///
+/// * `z` - Level of zoom
+/// * `x` - X coordinate
+/// * `y` - Y coordinate
+/// * `layer_name` - Name of the layer
+/// * `records` - Records to add as features to the MVT tile
+pub fn create_and_fill_mvt_tile(
+    z: u64,
+    x: u64,
+    y: u64,
+    layer_name: &str,
+    records: Vec<GeoJsonAndData>,
+) -> MvtTile {
+    let mut tile = MvtTile::new(4096);
+    let mut mvt_layer = tile.create_layer(layer_name);
+    let ts = tile.extent() as f64;
+    let transform = MapGrid::default()
+        .tile_transform(
+            TileId::new(
+                x.try_into().unwrap(),
+                y.try_into().unwrap(),
+                z.try_into().unwrap(),
+            )
+            .unwrap(),
+        )
+        .scale(ts, ts);
+    for record in records.into_iter() {
+        let mut feature = mvt_layer.into_feature(record.as_geom_data(transform));
+        add_tags_to_feature(&mut feature, record.data, String::new());
+        mvt_layer = feature.into_layer();
+    }
+    tile.add_layer(mvt_layer).unwrap();
+    tile
+}
+
+/// Creates an SQL query to get geo json data
+///
+/// # Arguments
+///
+/// * `infra_id` - Infra id for which data is required
+/// * `table_name` - Table containing the data
+/// * `view` - View containing info to get the data
+/// * `z` - Level of zoom
+/// * `x` - X coordinate
+/// * `y` - Y coordinate
+pub fn get_geo_json_sql_query(
+    infra_id: i64,
+    table_name: &str,
+    view: &View,
+    z: u64,
+    x: u64,
+    y: u64,
+) -> String {
+    format!(
+        "
+        WITH bbox AS (
+            SELECT TileBBox({z}, {x}, {y}, 3857) AS geom
+        )
+        SELECT ST_AsGeoJson(geographic) AS geo_json, 
+            {data_expr} {exclude_fields} AS data 
+        FROM {table_name} layer 
+            CROSS JOIN bbox 
+            {joins} 
+        WHERE layer.infra_id = {infra_id} 
+            {where_condition}
+            AND {on_field} && bbox.geom 
+            AND ST_GeometryType({on_field}) != 'ST_GeometryCollection'
+        ",
+        on_field = view.on_field,
+        data_expr = view.data_expr,
+        exclude_fields = &format!(
+            "{} {}",
+            if view.exclude_fields.is_empty() {
+                ""
+            } else {
+                "-"
+            },
```

Comment:
This is dirty. You can compute exclude fields outside so we avoid nested `format!`. We can also use a simple map:

```rust
&view
  .exclude_fields
  .iter()
  .map(|field| format!("-'{field}'"))
  .collect::<Vec<_>>()
  .join(" ")
```